cmd/algokey/sign.go

```diff
@@ -41,6 +43,7 @@ func init() {
 	signCmd.MarkFlagRequired("txfile")
 	signCmd.Flags().StringVarP(&signOutfile, "outfile", "o", "", "Transaction output filename")
 	signCmd.MarkFlagRequired("outfile")
+	signCmd.Flags().BoolVarP(&rekeyed, "rekeyed", "r", false, "Required if the sending account has been rekeyed and the signing key doesn't match the account address")
```

Comment:
Why is this option required? Isn't it sufficient to (below) check if `key.SignatureVerifier = stxn.Sender` and set `stxn.AuthAddr` if not?

Reply:
I'll go ahead and make it automatic and get rid of the flag. (I originally put the flag there so that I wouldn't change any existing algokey behavior, because I was worried that there might be some tests or obscure use cases that depend on algokey's existing behavior. But this seems pretty unlikely since a transaction signed by the wrong key with no authaddr set is guaranteed to be invalid anyway.)